src/makeInfinityAdder.js

```diff
@@ -1,12 +1,23 @@
 'use strict';
 
-/***
- *
- * @return {function}
- */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number !== undefined) {
+      sum += number;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
    } 
    const result = sum;

    sum = 0;

    return result;
```

try to avoid else where it is possible. it will increase readibility of code